src/main/java/io/supertokens/webserver/api/multitenancy/ListConnectionUriDomainsAPI.java

```diff
@@ -95,7 +96,18 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IO
                     JsonArray tenantsArray = new JsonArray();
                     for (TenantConfig tenantConfig : entry2.getValue()) {
                         JsonObject tenantConfigJson = tenantConfig.toJson(shouldProtect,
-                                storage, CoreConfig.PROTECTED_CONFIGS);
+                                storage, CoreConfig.PROTECTED_CONFIGS, getVersionFromRequest(req).lesserThan(SemVer.v5_1));
+
+                        if (getVersionFromRequest(req).lesserThan(SemVer.v5_1)) {
+                            tenantConfigJson.remove("useFirstFactorsFromStaticConfigIfEmpty");
+                            tenantConfigJson.get("thirdParty").getAsJsonObject().remove("useThirdPartyProvidersFromStaticConfigIfEmpty");
+                        }
+
+                        if (getVersionFromRequest(req).lesserThan(SemVer.v5_0)) {
+                            tenantConfigJson.remove("firstFactors");
+                            tenantConfigJson.remove("requiredSecondaryFactors");
+                        }
```

Comment:
you should make the toJson function only do this logic? Otherwise it is easy to miss doing this

Reply:
Done